README.md

```diff
@@ -2,5 +2,263 @@
+
+```objc
+for (NSUInteger i = 1; i <= 100; i++) { 
+    NSString *result = @"";
+    
+    if (i % 3 == 0 && i % 5 == 0) {
+        result = @"FizzBuzz";
+    }
+    else if (i % 3 == 0) {    // becomes an 'else if' statement
+        result = @"Fizz";
+    }
+    else if (i % 5 == 0) {
+        result = @"Buzz";
+    }
+    else {
+        result = [NSString stringWithFormat:@"%lu", i];
+    }
+    NSLog(@"%@", result);
+}
+```
+This should now print the correct sequence to FizzBuzz:
+
+```
+1
+2
+Fizz
+4
+Buzz
+Fizz
+7
+8
+Fizz
+Buzz
+11
+Fizz
+13
+14
+FizzBuzz
+...
+```
+7 — Now that we have a working (but **not** ideal) implementation of FizzBuzz, we can play around with the loop's parameters to get a different "slice" of the FizzBuzz sequence. While we could do this directly in the loop's conditional line, let's move the values into variables to follow good-practice guidelines of avoiding "magic numbers". Before the loop, declare two `NSUInteger` variables called `start` and `limit`. Set them equal to `1` and `100` respectively and replace the integers in the `for` loop's conditional line with these new variables.:
+
+```objc
+NSUInteger start = 1;
+NSUInteger limit = 100;
+
+for (NSUInteger i = start; i <= limit; i++) { 
+    NSString *result = @"";
+    
+    if (i % 3 == 0 && i % 5 == 0) {
+        result = @"FizzBuzz";
+    }
+    else if (i % 3 == 0) {    // becomes an 'else if' statement
+        result = @"Fizz";
+    }
+    else if (i % 5 == 0) {
+        result = @"Buzz";
+    }
+    else {
+        result = [NSString stringWithFormat:@"%lu", i];
+    }
+    NSLog(@"%@", result);
+}
+```
+This should still print the FizzBuzz sequence from `1` to `100`:
+
+```
+1
+2
+Fizz
+4
+Buzz
+Fizz
+7
+8
+Fizz
+Buzz
+11
+Fizz
+13
+14
+FizzBuzz
+...
+```
+8 — Now, play around with the values of `start` and `limit`. Try getting the slice of `100` to `130`:
+
+```objc
+NSUInteger start = 100;
+NSUInteger limit = 130;
+...
+```
+
+This should print:
+
+```
+Buzz
+101
+Fizz
+103
+104
+FizzBuzz
+106
+107
+Fizz
+109
+Buzz
+Fizz
+112
+113
+Fizz
+Buzz
+116
+Fizz
+118
+119
+FizzBuzz
+121
+122
+Fizz
+124
+Buzz
+Fizz
+127
+128
+Fizz
+Buzz
+```
+
+## Advanced
+
+1. The FizzBuzz exercise is primarily about best practice and efficiency. Can you identify some of the problems with the implementation of FizzBuzz that we wrote? For starters, it repeats itself, both in the conditional checks, and in the instructions. Can you come up with a better logical structure that doesn't repeat itself?
+
+2. Furthermore, while we did document the starting and ending points for the loop by saving the values into the `start` and `limit` variables, the remainder (`%`) calculations are still written with "magic numbers". Document these values by creating new variables called `fizz` and `buzz` that are set to `3` and `5` respectively.
```

Comment:
Hmm I don't like `fizz` and `buzz` as names. Maybe `fizzDivisor` and `buzzDivisor`?